vllm/test_utils.py

```diff
@@ -128,3 +137,69 @@
 ]
 
 MODEL_WEIGHTS_S3_BUCKET = "s3://vllm-ci-model-weights"
+
+DUMMY_LOGITPROC_ENTRYPOINT = "dummy_logitproc"
+
+DUMMY_LOGITPROC_FQCN = "vllm.test_utils:DummyLogitsProcessor"
+
+
+class DummyLogitsProcessor(LogitsProcessor):
+    """Fake logit processor to support unit testing and examples"""
+
+    def __init__(self, vllm_config: "VllmConfig", device: torch.device,
+                 is_pin_memory: bool):
+        self.req_info: dict[int, SamplingParams] = {}
+
+    def is_argmax_invariant(self) -> bool:
+        """Never impacts greedy sampling"""
+        return False
+
+    def update_state(self, batch_update: Optional[BatchUpdate]):
+        if not batch_update:
+            return
+
+        # Process added requests.
+        for _, index, params, _ in batch_update.added:
+            assert params is not None
+            if params.extra_args:
+                target_token = params.extra_args.get("target_token", None)
+            else:
+                target_token = None
+            self.req_info[index] = target_token
```

Comment:
I know it simplifies the swapping logic but I'm not sure we should store a `None` entry for every request. Some/most logits processors would be rarely used and would be better to avoid this overhead.

If we are storing Nones like that it would probably make more sense to be a list instead (dict is used assuming sparsity)

I know this is a dummy/example but the expectation is that it could be used as a template for others.